hax/hax/ha/ha.py

```diff
@@ -0,0 +1,64 @@
+# Copyright (c) 2022 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License,
+# or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License
+# for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing, please email
+# opensource@seagate.com or cortx-questions@seagate.com.
+
+import logging
+from hax.ha.event.node import NodeEvent
+from hax.ha.message_type.message_type import HealthMessage
+from hax.types import ObjT, Fid
+from hax.util import ConsulUtil
+from cortx.utils.conf_store import Conf
+
+
+class Ha():
+    resources = {ObjT.NODE: NodeEvent}
+    # currently only online is supported for node status
+    message_types = {ObjT.NODE: [HealthMessage]}
+
+    def __init__(self, util: ConsulUtil):
+        self.util = util
+
+    def send_event(self,
+                   resource_type: ObjT,
```

Comment:
We are not really making use of polymorphism here. So, As we have a base class for event type, similarly we need to have a base class for resource type. All the resource types will inherit from the base type, e.g. Node. `Ha.send_event()` must take generic resource type as a parameter and instead of searching for the corresponding resource in the list, invoke, `resource.create_event()` directly, that should invoke `Node.create_event()`.

Node.create_event() must return HaEvent instance of NodeEvent type and that should have enough information to send. HaEvent must define a `send()` method which will be implemented by NodeEvent. So, we just invoke event.send().

Reply:
Created ticket https://jts.seagate.com/browse/CORTX-31068 to address it